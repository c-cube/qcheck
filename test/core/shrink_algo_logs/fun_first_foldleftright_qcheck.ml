open QCheck

let prop_foldleft_foldright_uncurry_funfirst =
  Test.make ~name:"fold_left fold_right uncurried fun first"
    (triple
       (fun1 Observable.(pair int int) small_nat)
       small_nat
       (list small_nat))
    (Log.shrinks Print.(triple Fn.print int (list int)) @@
     fun (f,z,xs) ->
       List.fold_right (fun x y -> Fn.apply f (x,y)) xs z =
       List.fold_left (fun x y -> Fn.apply f (x,y)) z xs)

let rand = Random.State.make [|1234|];;
try Test.check_exn (*~rand*) prop_foldleft_foldright_uncurry_funfirst with Test.Test_fail _ -> ()
