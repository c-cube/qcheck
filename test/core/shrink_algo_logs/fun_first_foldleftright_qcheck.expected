fails ({(3, 2) -> 30; (5, 9) -> 1; (8, 94) -> 0; (0, 3) -> 19; (9, 30) -> 4; (1, 3) -> 8; (3, 5) -> 8; (94, 8) -> 2; _ -> 7}, 5, [9; 3; 94; 3])
holds ({(3, 2) -> 30; (5, 9) -> 1; (8, 94) -> 0; (0, 3) -> 19; _ -> 7}, 5, [9; 3; 94; 3])
holds ({(3, 2) -> 30; (5, 9) -> 1; (8, 94) -> 0; (94, 8) -> 2; _ -> 7}, 5, [9; 3; 94; 3])
holds ({(3, 2) -> 30; (5, 9) -> 1; (3, 5) -> 8; (94, 8) -> 2; _ -> 7}, 5, [9; 3; 94; 3])
holds ({(3, 2) -> 30; (1, 3) -> 8; (3, 5) -> 8; (94, 8) -> 2; _ -> 7}, 5, [9; 3; 94; 3])
holds ({(9, 30) -> 4; (1, 3) -> 8; (3, 5) -> 8; (94, 8) -> 2; _ -> 7}, 5, [9; 3; 94; 3])
fails ({(3, 2) -> 30; (5, 9) -> 1; (8, 94) -> 0; (0, 3) -> 19; (9, 30) -> 4; (1, 3) -> 8; _ -> 7}, 5, [9; 3; 94; 3])
holds ({(3, 2) -> 30; (5, 9) -> 1; (8, 94) -> 0; _ -> 7}, 5, [9; 3; 94; 3])
holds ({(3, 2) -> 30; (5, 9) -> 1; (1, 3) -> 8; _ -> 7}, 5, [9; 3; 94; 3])
holds ({(3, 2) -> 30; (9, 30) -> 4; (1, 3) -> 8; _ -> 7}, 5, [9; 3; 94; 3])
holds ({(0, 3) -> 19; (9, 30) -> 4; (1, 3) -> 8; _ -> 7}, 5, [9; 3; 94; 3])
holds ({(3, 2) -> 30; (5, 9) -> 1; (8, 94) -> 0; (0, 3) -> 19; (9, 30) -> 4; _ -> 7}, 5, [9; 3; 94; 3])
fails ({(3, 2) -> 30; (5, 9) -> 1; (8, 94) -> 0; (0, 3) -> 19; (1, 3) -> 8; _ -> 7}, 5, [9; 3; 94; 3])
holds ({(3, 2) -> 30; (5, 9) -> 1; _ -> 7}, 5, [9; 3; 94; 3])
holds ({(3, 2) -> 30; (1, 3) -> 8; _ -> 7}, 5, [9; 3; 94; 3])
holds ({(0, 3) -> 19; (1, 3) -> 8; _ -> 7}, 5, [9; 3; 94; 3])
holds ({(3, 2) -> 30; (5, 9) -> 1; (8, 94) -> 0; (0, 3) -> 19; _ -> 7}, 5, [9; 3; 94; 3])
holds ({(3, 2) -> 30; (5, 9) -> 1; (8, 94) -> 0; (1, 3) -> 8; _ -> 7}, 5, [9; 3; 94; 3])
holds ({(3, 2) -> 30; (5, 9) -> 1; (0, 3) -> 19; (1, 3) -> 8; _ -> 7}, 5, [9; 3; 94; 3])
holds ({(3, 2) -> 30; (8, 94) -> 0; (0, 3) -> 19; (1, 3) -> 8; _ -> 7}, 5, [9; 3; 94; 3])
fails ({(5, 9) -> 1; (8, 94) -> 0; (0, 3) -> 19; (1, 3) -> 8; _ -> 7}, 5, [9; 3; 94; 3])
holds ({(5, 9) -> 1; (8, 94) -> 0; _ -> 7}, 5, [9; 3; 94; 3])
holds ({(5, 9) -> 1; (1, 3) -> 8; _ -> 7}, 5, [9; 3; 94; 3])
holds ({(0, 3) -> 19; (1, 3) -> 8; _ -> 7}, 5, [9; 3; 94; 3])
holds ({(5, 9) -> 1; (8, 94) -> 0; (0, 3) -> 19; _ -> 7}, 5, [9; 3; 94; 3])
holds ({(5, 9) -> 1; (8, 94) -> 0; (1, 3) -> 8; _ -> 7}, 5, [9; 3; 94; 3])
holds ({(5, 9) -> 1; (0, 3) -> 19; (1, 3) -> 8; _ -> 7}, 5, [9; 3; 94; 3])
holds ({(8, 94) -> 0; (0, 3) -> 19; (1, 3) -> 8; _ -> 7}, 5, [9; 3; 94; 3])
fails ({(5, 9) -> 1; (8, 94) -> 0; (0, 3) -> 19; (1, 3) -> 8; _ -> 0}, 5, [9; 3; 94; 3])
holds ({(5, 9) -> 1; (8, 94) -> 0; _ -> 0}, 5, [9; 3; 94; 3])
holds ({(5, 9) -> 1; (1, 3) -> 8; _ -> 0}, 5, [9; 3; 94; 3])
fails ({(0, 3) -> 19; (1, 3) -> 8; _ -> 0}, 5, [9; 3; 94; 3])
fails ({(0, 3) -> 19; _ -> 0}, 5, [9; 3; 94; 3])
holds ({_ -> 0}, 5, [9; 3; 94; 3])
holds ({(0, 3) -> 0; _ -> 0}, 5, [9; 3; 94; 3])
fails ({(0, 3) -> 10; _ -> 0}, 5, [9; 3; 94; 3])
holds ({_ -> 0}, 5, [9; 3; 94; 3])
holds ({(0, 3) -> 0; _ -> 0}, 5, [9; 3; 94; 3])
fails ({(0, 3) -> 5; _ -> 0}, 5, [9; 3; 94; 3])
holds ({_ -> 0}, 5, [9; 3; 94; 3])
holds ({(0, 3) -> 0; _ -> 0}, 5, [9; 3; 94; 3])
fails ({(0, 3) -> 4; _ -> 0}, 5, [9; 3; 94; 3])
holds ({_ -> 0}, 5, [9; 3; 94; 3])
holds ({(0, 3) -> 0; _ -> 0}, 5, [9; 3; 94; 3])
fails ({(0, 3) -> 3; _ -> 0}, 5, [9; 3; 94; 3])
holds ({_ -> 0}, 5, [9; 3; 94; 3])
holds ({(0, 3) -> 0; _ -> 0}, 5, [9; 3; 94; 3])
fails ({(0, 3) -> 2; _ -> 0}, 5, [9; 3; 94; 3])
holds ({_ -> 0}, 5, [9; 3; 94; 3])
fails ({(0, 3) -> 1; _ -> 0}, 5, [9; 3; 94; 3])
holds ({_ -> 0}, 5, [9; 3; 94; 3])
holds ({(0, 3) -> 0; _ -> 0}, 5, [9; 3; 94; 3])
fails ({(0, 3) -> 1; _ -> 0}, 0, [9; 3; 94; 3])
holds ({_ -> 0}, 0, [9; 3; 94; 3])
holds ({(0, 3) -> 0; _ -> 0}, 0, [9; 3; 94; 3])
fails ({(0, 3) -> 1; _ -> 0}, 0, [94; 3])
holds ({_ -> 0}, 0, [94; 3])
holds ({(0, 3) -> 0; _ -> 0}, 0, [94; 3])
fails ({(0, 3) -> 1; _ -> 0}, 0, [3])
holds ({_ -> 0}, 0, [3])
holds ({(0, 3) -> 0; _ -> 0}, 0, [3])
holds ({(0, 3) -> 1; _ -> 0}, 0, [])
holds ({(0, 3) -> 1; _ -> 0}, 0, [0])
holds ({(0, 3) -> 1; _ -> 0}, 0, [2])
